source/types/Case.ts

```diff
@@ -0,0 +1,267 @@
+  /** Application is fully signed by all applicant and is ready to send in. */
+  | "active:signature:completed"
+  /** Application is not fully signed. One or more applicants need to sign this application. */
+  | "active:signature:pending"
+  /** Application has been submitted to the provider and is awaiting processing. */
+  | "active:submitted"
+  /** Application is closed. */
+  | "closed"
+  /** Application has not yet been started by the applicant. */
+  | "notStarted"
+
+  /** VIVA */
+  /** Application has been sent to VIVA and is awaiting action from a case worker. */
+  | "active:submitted:viva"
+  /** Completion ("stickprovskontroll") is required. */
+  | "active:completionRequired:viva"
+  /** Application is fully approved. Calculation/payment info is available. */
+  | "closed:approved:viva"
+  /** Completion ("stickprovskontroll") has been rejected. */
+  | "closed:completionRejected:viva"
+  /** Application is partially approved. Calcuation/payment info is available. */
+  | "closed:partiallyApproved:viva"
+  /** Application has been processed by a case worker and has been rejected. Rejection info is available. */
+  | "closed:rejected:viva"
+  /** Application has not yet been started by the applicant (VIVA-specific). */
+  | "notStarted:viva";
+
+/**
+ * Possible roles for a person on a case.
+ */
+export type PersonRole = "applicant" | "coApplicant" | "children";
+
+/**
+ * Administrator (case worker).
+ */
+export interface Administrator {
+  email: string;
+  name: string;
+  phone?: string;
+  title: string;
+}
+
+/**
+ * Represents a time period (with a start and end timestamp).
+ */
+export interface Period {
+  /**
+   * Unix timestamp in milliseconds for when this period ends.
+   */
+  endDate: number;
+
+  /**
+   * Unix timestamp in milliseconds for when this period starts.
+   */
+  startDate: number;
+}
+
+/**
+ * Details of a case as retrieved from VIVA.
+ */
+export interface VIVACaseDetails {
+  /**
+   * List of case workers associated with this case.
+   */
+  administrators: Administrator[];
+
+  /**
+   * The period this case pertains to.
+   */
+  period: Period;
+
+  /**
+   * Unique ID associated with this case in VIVA.
+   */
+  workflowId: string;
+}
+
+export interface AnswerField {
+  /**
+   * Unique ID for this field in the form.
+   */
+  id: string;
+
+  /**
+   * List of tags with generic use.
+   *
+   * For VIVA this is used to connect and transform answers to VIVA properties in VADA.
+   */
+  tags?: string[];
+}
+
+export interface Answer {
+  /**
+   * Details of the field for this answer.
+   */
+  field: AnswerField;
+
+  /**
+   * The actual value of this answer.
+   */
+  value: string | boolean;
+}
+
+export interface EncryptedAnswersWrapper {
+  /**
+   * Encrypted string output of the answers object.
+   */
+  encryptedAnswers: string;
+}
+
+export interface FormPosition {
+  currentMainStep: number;
+  currentMainStepIndex: number;
+  index: number;
+  level: number;
+}
```

Comment:
I think the same. What is this? :)

Reply:
I don't know. I haven't been able to wrap my head around the form position system yet, so I'm unsure what each individual property means.

Reply:
Good question. We leave it as it is now. Can add a task to address this in future refactoring.